src/core/metrics.cc

```diff
@@ -112,6 +112,15 @@ future<> configure(const options& opts, int handle) {
     });
 }
 
+future<>
+replicate_metrics(int source_handle,
+                  std::vector<impl::replicated_metric_spec> replication_spec) {
+    return smp::invoke_on_all([source_handle, replication_spec] {
+        auto source_impl = impl::get_local_impl(source_handle);
```

Comment:
Hold by ref?

Reply:
I'd rather hold by value for a `shared_ptr`. `const&` would work, but I don't like the lifetime extension stuff that comes along with it.

Reply:
I see it's returned by value